addons/mail/static/src/xml/thread.xml

```diff
@@ -312,9 +312,10 @@
                         <t t-esc="message.getDisplayedAuthor()"/>
                     </strong>
                     <strong t-elif="message.hasEmailFrom()">
-                        <a class="text-muted" t-attf-href="mailto:#{message.getEmailFrom()}?subject=Re: #{message.hasSubject() ? message.getSubject() : ''}">
+                        <a t-if="message.shouldRedirectToAuthor()" class="text-muted" t-attf-href="mailto:#{message.getEmailFrom()}?subject=Re: #{message.hasSubject() ? message.getSubject() : ''}">
```

Comment:
Is this change necessary? What is the case it is trying to cover?

`hasEmailFrom` is on an `elif` of which the corresponding `if` is `hasAuthor`, and `shouldRedirectToAuthor` will always return `true` if `hasAuthor` is `false` (through `_isMyselfAuthor`).

Reply:
@seb-odoo yes, because in social we have an `email_from` (with the Facebook user info), but we don't want to be able to click on the contact name in Discuss

Reply:
Ok for the reasoning, but the method `shouldRedirectToAuthor` still does not make sense if `hasAuthor` is `false`, it should never be called in that context. It's actually currently returning an incorrect value in that case, because it returns `true` but there is no author set, so to which author it should redirect?!

I think you need to find a condition that specifically work for your Facebook case (do you know it is a Facebook message?), and it should likely be an override in `social`.

Reply:
@seb-odoo It that case, we do not redirect to `mailto: <email_from>`, I can add a new method `shouldRedirectToEmailFrom`, but I think it's cleaner to use the same method.

(if an author exist, `shouldRedirectToAuthor` tell us if we redirect to the `res.partner`, otherwise, it tell us if we should open the email in the default mail client when clicking on the partner name))

Hope it's cear

Reply:
Having one method doing 2 things in 2 different contexts is not a good idea.

I would prefer a new method such as `shouldRedirectToEmailFrom`, but with another name, because it is not a "redirect" when it's an email, but it's a "mailto link". So maybe just `hasMailtoLink`.

Note that is it out of scope of this PR, but in livechat the mailto link is displayed even if no email address is set/it is invalid, so that method will also help avoid that case in the future.